### DISCUSSION TITLE
Override footer

### DISCUSSION THREAD

**Discussion by FernandoBasso** [pradyunsg/furo#248] [Questions] [answered]
I am trying to override the footer. I created `_templates/footer.html` with the following content:

```
{% extends '!layout.html' %}

{% block footer %}
<footer>
My Custom Footer
</footer>
{% endblock %}
```

I see no errors when I restart the server, but nothing changes on the footer. I also tried extending `page.html` and even `base.html`. Nothing worked. Never saw any errors on the server output.

`conf.py` has a line `templates_path = ['_templates']`.

Is it possible and I am just doing it incorrectly or it is really not possible?

**Comment by pradyunsg** [comment]
Furo doesn't have a footer.html file. You'll need to override `layout.html`. That should be basically the same code, but with the filename changed.

Notes in https://pradyunsg.me/furo/customisation/injecting/#html and https://pradyunsg.me/furo/stability/#undocumented-unstable-customisations apply.

**Comment by FernandoBasso** [comment, marked as answer]
Thanks for the reply. I think I am still doing something wrong.

```
$ cat docs/_templates/layout.html 
{% extends "layout.html" %}

{% block footer %}
<footer>
My Custom Footer
</footer>
{% endblock %}

$ grep templates_path docs/conf.py 
templates_path = ['_templates']
```
I also tried changing `{% extends "layout.html" %}` with `{% extends "furo/layout.html" %}` and also prefixing it with the exclamation mark "!". No errors, nothing changes.

[Sphinx documentation](https://www.sphinx-doc.org/en/master/development/theming.html#templating) says:

> When extending a template in the base theme with the same name, use the theme name as an explicit directory: `{% extends "basic/layout.html" %}`. From a user `templates_path` template, you can still use the “exclamation mark” syntax as described in the templating document.

In [Templating](https://www.sphinx-doc.org/en/master/templating.html), it reads:

> Sphinx’s basic theme provides base templates with a couple of blocks it will fill with data. These are located in the themes/basic subdirectory of the Sphinx installation directory, and used by all builtin Sphinx themes. Templates with the same name in the templates_path override templates supplied by the selected theme.

I can't figure out how to translate those explanations to Furo's context. I'm failing to understand why the approaches I have tried do not even produce an error or warning. As your answer suggests, it should be straightforward. What am I doing wrong?

**Reply by pradyunsg** [reply]
Try passing `-a` to the sphinx-build command, so that it doesn't attempt an incremental build.

**Reply by FernandoBasso** [reply]
Passing `-a` and nothing changes. I even tried removing `docs/_build/`.

**Reply by pradyunsg** [reply]
Ah buggery. It's page.html not layout.html.

**Reply by FernandoBasso** [reply]
Yes, it works! Thanks a lot.

EDIT: By the way, I do not want (and will not) remove the footer's attribution to @pradyunsg and the Furo theme. On the contrary, I want to advertise this awesome theme ❤️. I just want to add licensing information with links, which seems not to be possible from `conf.py`.
